scripts/run-metadata-validation.ps1

```diff
@@ -0,0 +1,48 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+<#
+.Synopsis
+    Runs MetadataParser tool to validate latest XSLT rules don't break downstream parsing of the metadata
+
+.Description
+    1. tranforms latests snapshots of beta and v1 metadata
+    2. validates that the transformed metadata is parsable as OData EDM model and conversion to OpenAPI document is possible
+
+.Example
+    ./scripts/run-metadata-validation.ps1 -repoDirectory C:/github/msgraph-metadata
+
+.Example
+    ./scripts/run-metadata-validation.ps1 -repoDirectory $GITHUB_WORKSPACE
+
+.Parameter repoDirectory
+    Full path the the root directory of msgraph-metadata checkout.
+#>
+
+param(
+    [Parameter(Mandatory=$true)][string]$repoDirectory
+)
+
+$transformCsdlDirectory = Join-Path $repoDirectory "transforms/csdl"
+$transformScript = Join-Path $transformCsdlDirectory "transform.ps1"
+$xsltPath = Join-Path $transformCsdlDirectory "preprocess_csdl.xsl"
+
+$betaSnapshot = Join-Path $repoDirectory "beta_metadata.xml"
+$v1Snapshot = Join-Path $repoDirectory "v1.0_metadata.xml"
+
+$metadataParserTool = Join-Path $repoDirectory "tools/MetadataParser/MetadataParser.csproj"
+
+$transformedBeta = Join-Path $repoDirectory "transformed_beta_metadata.xml"
+$transformedV1 = Join-Path $repoDirectory "transformed_v1.0_metadata.xml"
+
+Write-Host "Tranforming beta metadata using xslt..."
+& $transformScript -xslPath $xsltPath -inputPath $betaSnapshot -outputPath $transformedBeta
+
+Write-Host "Validating beta metadata after the transform..."
+& dotnet run -p $metadataParserTool $transformedBeta
```

Comment:
nit: I'd move the dotnet run steps in the workflow, this way we'd have a better integration for linting, warning etc... with the UI

Reply:
I would like to be able to run this script as a self contained test in local dev box without needing to create a PR and then realizing that my XSLT is wrong. That is the reason I am keeping everything in a single ps1 file. I wish Actions had a way to run as is on a local dev environment without much modifications.

Reply:
yeah not having the ability to run workflows locally is a real pain. I pushed 18 commits to fix the workflow before squashing and force push....

you could have two scripts:
- one that calculates the paths and settings them as tasks output (this one
- one that calls the first one and dotnet run for the local debug experience

Reply:
I think this is a script that is short enough to be single file. And local testing instruction is as simple as:
> After you made your changes to the XSLT, run:
> `./scripts/run-metadata-validation.ps1 <local_metadata_checkout_folder>`

And everything is maintained in a single place.